### PR TITLE
feat: advancedNetwork, interceptors, standalone

### PR DESCRIPTION
**Pull request by millcek**

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -1,17 +1,55 @@
...
+			response?: RequestOrResponse
+			url: string
+		},
+		context: { pageUrl: string },
+	) => void | boolean
+	url?: (data: { url: string }) => void
+}
+
+type InitParams = {
+	// Advanced network recording allows recording the bodies and headers of requests and responses.
```
Comment (Joozty) [inline index.ts:27]: these descriptions should go to README.md as well... ideally add there also links to appropriate pages to our docs.

File: index.ts [hunk 2]
```diff
@@ -1,17 +1,55 @@
...
+
+interface Interceptors {
+	network?: (
+		data: {
+			request?: RequestOrResponse
+			response?: RequestOrResponse
+			url: string
+		},
+		context: { pageUrl: string },
+	) => void | boolean
```
Comment (amertak) [inline index.ts:22]: void | false

**Review by amertak — CHANGES_REQUESTED** [review]
Nahod to prosim do toho readme.md jeste

Upravim to v readme.com to cele smartlook init volani o ty chybejici parametry

**Review by Joozty — APPROVED** [review]
👍

**Comment by Joozty** [comment]
Actually I just noticed, you forgot to update init params 

https://github.com/smartlook/smartlook-client/blob/feat/advancedNetwork%2C-interceptors%2C-standalone/index.ts#L66

File: index.ts [hunk 3]
```diff
@@ -1,17 +1,55 @@
...
+			request?: RequestOrResponse
+			response?: RequestOrResponse
+			url: string
+		},
+		context: { pageUrl: string },
+	) => void | false
+	url?: (data: { url: string }) => void
+}
+
+type InitParams = {
```
Comment (Joozty) [inline index.ts:26]: this and maybe above interface as well coudl get exported so customer can use it like this:

```ts
const interceptors: Interceptors = ...
const config: InitParams = {..., interceptors}
smartlook.init('key', config)
```

what do you think @smartlook/web-sdk?

**Comment by millcek** [comment]
pushed to master directly

**Pull request closed without merging**